tests/zfs-tests/tests/functional/mmp/mmp_thread_basic.ksh

```diff
@@ -0,0 +1,91 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2017 by Lawrence Livermore National Security, LLC.
+#
+
+# DESCRIPTION:
+#
+# STRATEGY:
+#	1. Set zfs_mmp_interval to 0 (disables mmp)
+#	2. Set zfs_txg_timeout to large value
+#	3. Create a zpool
+#	4. Force a sync on the zpool
+#	5. Store the current "best" uberblock
+#	6. Repeatedly check that the "best" uberblock
+#	   for 10 seconds
+#	7. If the uberblock never changed, fail
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+verify_runnable "both"
+TXG_TIMEOUT=
+
+function cleanup
+{
+	set_tunable64 zfs_txg_timeout 5
+
+	if poolexists mmptestpool; then
+		log_must zpool destroy mmptestpool
+	fi
+
+	log_must rm -rf "$TEST_BASE_DIR/mmp_vdevs"
+	log_must rm -f $PREV_UBER $CURR_UBER
+}
+
+log_assert "mmp thread writes uberblocks (MMP)"
+log_onexit cleanup
+
+log_must mkdir "$TEST_BASE_DIR/mmp_vdevs"
```

Comment:
@ofaaland There wasn't really a reason. I just felt more comfortable creating a vdev file I knew was clean and untouched. It's not expensive to do so with `truncate` also. If you feel it's better to use `DISK0`, then we can remove the creation of the vdev file.